neurow/lib/neurow/internal_api/publish_request.ex

```diff
@@ -0,0 +1,120 @@
+defmodule Neurow.InternalApi.PublishRequest do
+  alias Neurow.InternalApi.Message
+
+  defstruct [:topics, :topic, :message, :messages]
+
+  def from_json(payload) when is_binary(payload) do
+    case Jason.decode(payload) do
+      {:ok, decoded_payload} -> {:ok, from_json(decoded_payload)}
+      error -> error
+    end
+  end
+
+  def from_json(payload) when is_map(payload) do
+    %Neurow.InternalApi.PublishRequest{
+      topic: payload["topic"],
+      topics: payload["topics"],
+      message:
+        case payload["message"] do
+          message when is_map(message) ->
+            Neurow.InternalApi.Message.from_json(payload["message"])
+
+          _ ->
+            nil
```

Comment:
nil -> nil, but _ -> error?

Reply:
The validation is performed later, so I prefer not to return an error at this step.